pkg/apis/extensions/v1beta1/types.go

```diff
@@ -1036,3 +1037,67 @@ type PodSecurityPolicyList struct {
 	// Items is a list of schema objects.
 	Items []PodSecurityPolicy `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   TemplateSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status TemplateStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty" protobuf:"bytes,2,rep,name=parameters"`
+
+	// Required: A list of resources to create
+	Objects []runtime.RawExtension `json:"objects,omitempty" protobuf:"bytes,3,rep,name=objects"`
+
+	// Optional: ObjectLabels is a set of labels that are applied to every
```

Comment:
Conversely, why not let people specify the same parameter for the labels in every object?